airbyte-integrations/connectors/source-marketo/source_marketo/utils.py

```diff
@@ -44,29 +44,30 @@ def clean_string(string: str) -> str:
 
 
 def format_value(value, schema):
-    if not isinstance(schema["type"], list):
-        field_type = [schema["type"]]
-    else:
-        field_type = schema["type"]
+    """Need use try...except construction for condition
+    when we get unexpected type from API"""
 
-    if value in [None, "", "null"]:
-        return None
-    elif "integer" in field_type:
-        if isinstance(value, int):
-            return value
+    try:
+        if not isinstance(schema["type"], list):
+            field_type = [schema["type"]]
+        else:
+            field_type = schema["type"]
 
-        # Custom Marketo percent type fields can have decimals, so we drop them
-        decimal_index = value.find(".")
-        if decimal_index > 0:
-            value = value[:decimal_index]
-        return int(value)
-    elif "string" in field_type:
-        return str(value)
-    elif "number" in field_type:
-        return float(value)
-    elif "boolean" in field_type:
-        if isinstance(value, bool):
-            return value
-        return value.lower() == "true"
+        if value in [None, "", "null"]:
+            return None
+        elif "integer" in field_type:
+            if isinstance(value, int):
+                return value
+            return int(value)
+        elif "string" in field_type:
+            return str(value)
+        elif "number" in field_type:
+            return float(value)
+        elif "boolean" in field_type:
+            if isinstance(value, bool):
+                return value
+            return value.lower() == "true"
 
-    return value
+        return value
+    except ValueError:
```

Comment:
Some generic thoughts on the issue of trying to convert float to boolean: as far as I can tell from reading the code, this means we are inferring the schema wrong _somewhere_, right? we've already verified that `number` types are correct in the `leads` stream, so it must mean the issue is somewhere in the acitivities streams right? unless we think hubspot is incorrectly declaring the schema in which case, we should print exactly the stream name which is failing. (see point 2 below) 

Some specific thoughts on this line: 

1. I'm not sure this is the right solution -- if we have an exception, this means the declared schema for this column is wrong, and it will cause a failure in the destination, right? if yes, as a stop gap we should return `None` and ask the user to upgrade to the newer version and help us see what the incorrect value is
2. we should add a log line in the CDK that indicates exactly which stream failed if an exception is thrown while reading it. Right now we are guessing that it's the `leads` stream which is failing but i noticed the "activities" stream dynamically sets stream types, so that could be the issue
3. we should log the failing record here to understand exactly what the issue is

Reply:
For now, I think we can merge the solution that we have (set to values string if they failed typing and log it). And ask the contributor to test it with his values and after that, we can understand exactly what the issue is.

Reply:
if we reach the `catch` block, this would mean that the value does not match the declared schema and therefore will cause a breakage downstream no? I would rather we just throw the exception directly in marketo connector, at least we don't incur a lot of wasted time on the destination

Reply:
No, for leads stream I add in scheme date type `string`, so we pasted to destination string value (e.g. 'false'). And sync should be successful finished, but in log we have info about value, field, and schema data types.